### ISSUE TITLE
Github Desktop Flickering Uncontrollably 

### ISSUE THREAD

**Issue by HilalHakla** [desktop/desktop#6670]
## Description
![20190119_215108](https://user-images.githubusercontent.com/26529048/51431035-9f6b5780-1c34-11e9-9971-d39a1203c02b.gif)

## Version
GitHub Desktop: 1.6

* Operating system:
Microsoft Windows [Version 10.0.15063]
Windows 10

## Steps to Reproduce
Just Opening the IDE after installation

### Actual Behavior
Should not flicker

### Logs
2019-01-19 21:46:04> Program: Starting Squirrel Updater: --createShortcut GitHubDesktop.exe -l StartMenu,Desktop
2019-01-19 21:46:04> ApplyReleasesImpl: About to create shortcuts for GitHubDesktop.exe, rootAppDir C:\Users\Hilal\AppData\Local\GitHubDesktop
2019-01-19 21:46:04> ApplyReleasesImpl: Creating shortcut for GitHubDesktop.exe => C:\Users\Hilal\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\GitHub, Inc.\GitHub Desktop.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: About to save shortcut: C:\Users\Hilal\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\GitHub, Inc.\GitHub Desktop.lnk (target C:\Users\Hilal\AppData\Local\GitHubDesktop\GitHubDesktop.exe, workingDir C:\Users\Hilal\AppData\Local\GitHubDesktop\app-1.6.0, args , toastActivatorCSLID 27d44d0c-a542-5b90-bcdb-ac3126048ba2)
2019-01-19 21:46:04> ApplyReleasesImpl: Creating shortcut for GitHubDesktop.exe => C:\Users\Hilal\Desktop\GitHub Desktop.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: About to save shortcut: C:\Users\Hilal\Desktop\GitHub Desktop.lnk (target C:\Users\Hilal\AppData\Local\GitHubDesktop\GitHubDesktop.exe, workingDir C:\Users\Hilal\AppData\Local\GitHubDesktop\app-1.6.0, args , toastActivatorCSLID 27d44d0c-a542-5b90-bcdb-ac3126048ba2)
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: File Explorer.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: Git for Windows.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: Google Chrome.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: Microsoft SQL Server Management Studio 18.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: Outlook.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: Remote Desktop Connection.lnk
2019-01-19 21:46:04> ApplyReleasesImpl: Examining Pin: Visual Studio 2017.lnk
2019-01-19 21:46:08> Program: Starting Squirrel Updater: --checkForUpdate https://central.github.com/api/deployments/desktop/desktop/latest?version=1.6.0&env=production
2019-01-19 21:46:08> Program: Fetching update information, downloading from https://central.github.com/api/deployments/desktop/desktop/latest?version=1.6.0&env=production
2019-01-19 21:46:08> SingleGlobalInstance: Failed to grab lockfile, will retry: C:\Users\Hilal\AppData\Local\Temp\.squirrel-lock-85AC6050841D040A2B625FD633FF30F5BACB1066: System.IO.IOException: The process cannot access the file 'C:\Users\Hilal\AppData\Local\Temp\.squirrel-lock-85AC6050841D040A2B625FD633FF30F5BACB1066' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.SingleGlobalInstance..ctor(String key, TimeSpan timeOut)
2019-01-19 21:46:08> CheckForUpdateImpl: Generated new staging user ID: d5dabca9-5685-575f-8262-91238880db84
2019-01-19 21:46:09> CheckForUpdateImpl: Downloading RELEASES file from https://central.github.com/api/deployments/desktop/desktop/latest?version=1.6.0&env=production
2019-01-19 21:46:09> FileDownloader: Downloading url: https://central.github.com/api/deployments/desktop/desktop/latest/RELEASES?version=1.6.0&env=production&id=GitHubDesktop&localVersion=1.6.0&arch=amd64

**Comment by shiftkey** [comment]
@HilalHakla I've tried to reproduce this myself without any luck, using a single repository in the app but then I rename the directory to make it "missing".

Could you attach the full log files? The **Help** | **Show Logs in Explorer** menu items should open the logs folder, and inside those are the log files generated by day. This will provide some more information to help troubleshoot.

**Comment by HilalHakla** [comment]
OK  i solved it by 
1  - Deleting the "C:\Users\{Your User Name}\AppData\Local\GitHubDesktop"
2 - Uninstalling Github Desktop
3 - Removing any repository (or./and any .git files) that was opened in Github Desktop (Cause i think there was an issue with the repository that i opened)
4- Reinstall Github Desktop

**Comment by shiftkey** [comment]
@HilalHakla I'm glad you were able to workaround the issue. Is there anything interesting in the log files that might help us with reproducing the issue on our end?

**Comment by billygriffin** [comment]
@HilalHakla In addition to what @shiftkey asked above, I'm curious if 1.6.1 that landed today resolved this issue?

**Comment by x6herbius** [comment]
I was in the middle of writing a bug report for this issue myself, but found this page in the process.

https://imgur.com/a/2W7EbtL

I got it to occur while attempting to convert a Mercurial repo to a Git one:

1. Created an empty Git repository in the usual way with `git init`.
2. Opened the repo with GitHub Desktop.
2. Attempted to push from the Mercurial repository into this new repo folder, as per the `hg-git` instructions.
3. Realised that the push failed because I should have used `git init --blank`.
4. Removed the repo using `rmdir /s`
5. Created a new repo under the same name using `git init --blank`.
6. Retried the Mercurial push, which succeeded.

The main window then completely broke. Maybe this'll be of help to someone?

**Comment by billygriffin** [comment]
@desktop/support and @desktop/quality-assurance - Are either of y'all able to repro this so we can figure out what might be happening? My guess is that the mercurial -> Git use case isn't one that happens frequently enough that we should spend a ton of time trying to solve for it, but I'd love to understand if there's a broader issue here.

**Comment by tierninho** [comment]
@billygriffin No luck in reproducing the flickering. 

So far three reports of this, all on Windows:
- mercurial push to Git
- IDE was opened
- unknown reasons

One common thread is all the repos appear to be not published. The flickering is between the 'pre-publish repo' state and the 'cannot find repo' state. The aforementioned PR (#6228 ) looks to be inline with this.

**Comment by billygriffin** [comment]
I think we're probably going to see this happen more frequently before we can realistically diagnose what might be causing it. I'm sorry folks - I realize that's not super helpful for your case, but please let us know if you're continuing to see it and repro steps. In the mean time I'm going to close this issue to indicate that we don't intend to do any further work trying to get to the bottom of it.

**Comment by SethQuinnShort** [comment]
I am getting this same problem when using Unreal Engine 4 and I tried the uninstall method above and I still get the same flicker results. This happens each time I tried uninstalling/deleting files

**Comment by MishaelBoss** [comment]
hello, I also have a problem: all windows flicker. On Version 3.3.14 (x64)

**Comment by steveward** [comment]
@MishaelBoss this issue is a few years old, so if you're having problems please open a new issue and fill out the template so we can investigate further. Thanks!

**Comment by sybrew** [comment]
Experiencing the same since last week. It's indicative of a Gsync issue. Please investigate.

**Comment by chronvas** [comment]
Experiencing the same issue on a FreeSync monitor with an Nvidia graphics card

**Comment by HEELcaT666** [comment]
> Experiencing the same issue on a FreeSync monitor with an Nvidia graphics card

Just experienced this as well on a new installation, also on Gsync monitor + NVDA gpu

**Comment by ghostinthecamera** [comment]
Also experience this - turning Gsync off resolved the issue.

Nvidia GPU, Windows 11
